homeassistant/components/xiaomi_miio/const.py

```diff
@@ -198,10 +198,12 @@ class SetupException(Exception):
 
 # TODO: use const from pythonmiio once new release with the constant has been published. # pylint: disable=fixme
 ROCKROBO_S4 = "roborock.vacuum.s4"
+ROCKROBO_S4_MAX = "roborock.vacuum.a19"
 ROCKROBO_S5_MAX = "roborock.vacuum.s5e"
 MODELS_VACUUM = [
```

Comment:
I'd replace this with the upstream `SUPPORTED_MODELS` list (https://github.com/rytilahti/python-miio/pull/1170/files#diff-556a967d2df04cf4ce4702ba6447afbbe0451825cdf08a28a390c29e8731934eR132). See TODO in L199.

Reply:
I'm the one that added that TODO :) 

The latest published release of the lib does not include SUPPORTED_MODELS yet. Hence the TODO and copy pasting the constants.

Reply:
Also the references upstream PR has not been merged yet and will take a while to get merged and released. Rather quickly fix the user facing issue rather then wait for new lib release with just a constant.

Reply:
That's why I said "if" this requires a release :)

Reply:
aah mybad. Nah it does not require a release to work.